File: beautifulsoup/sigir_officers_volunteers.py
import json

import requests
from bs4 import BeautifulSoup

SIGIR_TEAM_MEMBERS_URL = "https://sigir.org/general-information/officers-and-volunteers/"


class SigirTeamMember:
    def __init__(self, name: str, position: str, email: str):
        self.name = name
        self.position = position
        self.email = email

    def __str__(self):
        return f"{self.name}, {self.position}, {self.email}"


def sigir_officers_volunteers() -> list[SigirTeamMember]:
    print("Scraping Sigir team members...")

    page = requests.get(SIGIR_TEAM_MEMBERS_URL)
    soup = BeautifulSoup(page.content, "html.parser", from_encoding="utf-8")

    member_p_list = soup.find_all("div", class_="entry-content")[0].find_all("p")

    members: list[SigirTeamMember] = []

    for member_p in member_p_list:
        name = member_p.find_next("a").text.strip()
        position = member_p.find_next("strong").text.strip()

        email = next((x for x in map(lambda a: a.text.strip(), member_p.contents) if '@' in x), "")
        members.append(SigirTeamMember(name, position, email))

    return members


def main():
    team_members = sigir_officers_volunteers()

    json_dict_array = []
    print("Found %s sigir team members" % len(team_members))
    for team_member in team_members:
        print(team_member)
        json_dict = {
            "name": team_member.name,
            "position": team_member.position,
            "email": (team_member.email if team_member.email != "" else None)
        }
        json_dict_array.append(json_dict)
    print(json.dumps(json_dict_array))


if __name__ == '__main__':
    main()

File: jsoup/src/main/kotlin/SigirOfficersVolunteers.kt
import org.json.JSONArray
import org.json.JSONObject
import org.jsoup.Jsoup

const val SIGIR_TEAM_MEMBERS_URL = "https://sigir.org/general-information/officers-and-volunteers/"

data class SigirTeamMember(val name: String, val position: String, val email: String)

fun main() {
    val teamMembers = sigirOfficersVolunteers()

    val jsonObjects = teamMembers.map { member ->
        JSONObject().put("name", member.name)
            .put("position", member.position)
            .put("email", member.email)
    }
    val jsonArray = JSONArray(jsonObjects)
    println(jsonArray)
}

fun sigirOfficersVolunteers(): List<SigirTeamMember> {
    println("Scraping Sigir team members...")

    val doc = Jsoup.connect(SIGIR_TEAM_MEMBERS_URL).get()

    val memberPList = doc.select("div.entry-content").first()?.select("p") ?: return emptyList()

    val members = mutableListOf<SigirTeamMember>()

    for (memberP in memberPList) {
        val name = memberP.selectFirst("a")?.text() ?: ""
        val position = memberP.selectFirst("strong")?.text() ?: ""
        val email = memberP.textNodes().find { it.text().contains("@") }?.text()?.trim() ?: ""
        members.add(SigirTeamMember(name, position, email))
    }

    return members
}
